src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -19,9 +18,46 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
+            string fxName = null;
+            List<Packaging.Core.PackageDependency> packages = new List<Packaging.Core.PackageDependency>();
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
+                {
+                    if (reader.Value.Equals(JsonProperties.Dependencies))
+                    {
+                        // Dependencies are stored in an array
+                        while (reader.Read() && reader.TokenType != JsonToken.EndArray)
+                        {
+                            string id = null;
+                            string version = null;
+                            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+                            {
+                                if (reader.TokenType.Equals(JsonToken.PropertyName) && reader.Value.Equals(JsonProperties.PackageId))
+                                {
+                                    reader.Read();
+                                    id = reader.Value.ToString();
+                                }
```

Comment:
Should something check if `reader.Read()` returns `false`?  
```suggestion
                                if (reader.TokenType.Equals(JsonToken.PropertyName) && reader.Value.Equals(JsonProperties.PackageId) && reader.Read())
                                {
                                    id = reader.Value.ToString();
                                }
```

Reply:
At this point we're looking at a PropertyName, so if the reader returns false, that means we have a malformed json object. I forget how newtonsoft.json handles that situation 😅

Reply:
It also kind of doesn't matter, because the only time `reader.Read()` will return false is when there's no more data in the stream, which also means there's no `EndObject` token, which means the parsing should throw an exception.

If there isn't already a test for error handling when there are invalid json syntax, then there should be.